packages/client/src/runtime/utils/validatePrismaClientOptions.ts

```diff
@@ -50,18 +60,26 @@ It should have this form: { url: "CONNECTION_STRING" }`,
       }
     }
   },
-  adapter: (_options: any) => {
-    return
+  adapter: (adapter, config) => {
+    if (adapter === undefined) {
```

Comment:
At some point me and Joël discussed if stricter validation for `adapter` makes things safer:

1. Always throw on explicit `undefined` value, but not on missing property.
2. Allow `null` for explicitly skipping driver adapter.

Rationale for that was that if `adapter` is `undefined` by accident, it many cases it is not really obvious that adapter is not used. We ourselves have been burned by that couple of times. Doing validation this way prevents this mistake, and those who want to use adapter conditionally can still pass `null`.

What would you think about that?

Reply:
Absolutely, will do this in a follow-up PR
https://github.com/prisma/team-orm/issues/388